packages/gatsby/index.d.ts

```diff
@@ -156,7 +156,11 @@ export type PageProps<
 /**
  * A props object passed into the Head function for [Gatsby Head API](https://gatsby.dev/gatsby-head).
  */
-export type HeadProps<DataType = object, PageContextType = object> = {
+export type HeadProps<
+  DataType = object,
+  PageContextType = object,
+  ServerDataType = object
+> = {
```

Comment:
Setup copied from `PageProps`: https://github.com/gatsbyjs/gatsby/blob/fe65c295952f49345dc9e4a3b8da49b674b198ff/packages/gatsby/index.d.ts#L87-L154

Reply:
With Head the location will only ever be `pathname` so there won't be a need for `LocationState` generic

So we can skip that